src/node/genesis_gen.h

```diff
@@ -104,15 +103,33 @@ namespace ccf
       auto member_cert_der =
         tls::make_verifier({pem.data(), pem.data() + pem.size()})
           ->der_cert_data();
-      auto member_id = mc->get(member_cert_der);
 
-      // Cert should be unique
+      auto member_id = mc->get(member_cert_der);
       if (member_id.has_value())
       {
         throw std::logic_error(fmt::format(
           "Member certificate already exists (member {})", member_id.value()));
       }
 
+      MemberStatus member_status = MemberStatus::ACCEPTED;
+      auto service_status = get_service_status();
+      if (!service_status.has_value())
+      {
+        throw std::logic_error("Failed to get active service");
+      }
+
+      // If the service is opening, members are added as ACTIVE
```

Comment:
Is that right? Do we then reset their status after opening?

It wouldn't be right for a member to be added who has neither voted on opening nor acked to be counted as active.

Reply:
That's a very good point. We don't reset their status after opening the service indeed. 

I see two options:
- All members are added as `ACCEPTED` and `ACCEPTED` members can propose/vote when the service is opening. The members who vote for the service to open automatically become `ACTIVE` when the service is open - other members stay `ACCEPTED`. 
- All members are added as `ACTIVE` when the service is opening. When the service is opening, the members who haven't voted for the proposal are reverted to `ACCEPTED`.

Reply:
I am starting to wonder if a single value is a good way to capture this.

Members can:

1. Have endorsed the ledger up to a point (vote in opening, ack)
2. Vote
3. Count towards a vote count
4. Use their keyshare to execute a recovery

Reply:
Having read the relevant parts of the TR and thought about this a bit, I don't think the service status should have a bearing on this.

```
ACCEPTED - Ack (no exceptions) -> ACTIVE - Removal proposal passes -> RETIRED
```

This should be the state machine at all times. Members can never be automatically ACTIVE, because it would wrongly imply their participation, and it would count towards quorum, potentially stalling votes. That's true all the same when OPENING.

So I suggest a third option, which is that members have to Ack before they can vote/be counted.

If we want to avoid unnecessary acking, I think the right solution is to allow ACCEPTED members to always vote, and count the first vote as an ack. This is problematic though, for them to vote, they must be counted, and counting identities that never have shown activity or signed a single thing with their private key seems overly optimistic.

Reply:
@jumaffre as discussed, let's stick with the TR approach for now, which has the merit of being simple to explain and to implement. Only ACTIVE members can vote. Members become ACTIVE on first ack.

Reply:
Raised https://github.com/microsoft/CCF/issues/1113. Merging this PR now as it does not change what will be addressed in #1113.